include/mbedtls/md.h

```diff
@@ -132,220 +142,266 @@ void mbedtls_md_free( mbedtls_md_context_t *ctx );
  *
- * \returns         0 on success, MBEDTLS_ERR_MD_BAD_INPUT_DATA if parameter
- *                  verification fails.
+ * \param ctx       The generic message-digest context.
+ *
+ * \returns         \c 0 on success, #MBEDTLS_ERR_MD_BAD_INPUT_DATA if 
+ *                  parameter verification fails.
  */
 int mbedtls_md_starts( mbedtls_md_context_t *ctx );
 
 /**
- * \brief           Generic message digest process buffer
- *                  Called between mbedtls_md_starts() and mbedtls_md_finish().
+ * \brief           This function feeds an input buffer into an ongoing 
+ *                  message-digest computation.
+ *
+ *                  It is called between mbedtls_md_starts() and 
+ *                  mbedtls_md_finish().
  *                  May be called repeatedly.
  *
- * \param ctx       Generic message digest context
- * \param input     buffer holding the  datal
- * \param ilen      length of the input data
+ * \param ctx       The generic message-digest context.
+ * \param input     The buffer holding the input data.
+ * \param ilen      The length of the input data.
  *
- * \returns         0 on success, MBEDTLS_ERR_MD_BAD_INPUT_DATA if parameter
- *                  verification fails.
+ * \returns         \c 0 on success, #MBEDTLS_ERR_MD_BAD_INPUT_DATA if 
+ *                  parameter verification fails.
  */
 int mbedtls_md_update( mbedtls_md_context_t *ctx, const unsigned char *input, size_t ilen );
 
 /**
- * \brief           Generic message digest final digest
- *                  Called after mbedtls_md_update().
- *                  Usually followed by mbedtls_md_free() or mbedtls_md_starts().
+ * \brief           This function finishes the digest operation, 
+ *                  and writes the result to the output buffer.
  *
- * \param ctx       Generic message digest context
- * \param output    Generic message digest checksum result
+ *                  It is called after mbedtls_md_update(). It is usually 
+ *                  followed by mbedtls_md_free() or mbedtls_md_starts().
  *
- * \returns         0 on success, MBEDTLS_ERR_MD_BAD_INPUT_DATA if parameter
- *                  verification fails.
+ * \param ctx       The generic message-digest context.
+ * \param output    The buffer for the generic message-digest checksum result.
+ *
+ * \returns         \c 0 on success, or #MBEDTLS_ERR_MD_BAD_INPUT_DATA if 
+ *                  parameter verification fails.
  */
 int mbedtls_md_finish( mbedtls_md_context_t *ctx, unsigned char *output );
 
 /**
- * \brief          Output = message_digest( input buffer )
- *
- * \param md_info  message digest info
- * \param input    buffer holding the  data
- * \param ilen     length of the input data
- * \param output   Generic message digest checksum result
- *
- * \returns        0 on success, MBEDTLS_ERR_MD_BAD_INPUT_DATA if parameter
- *                 verification fails.
+ * \brief          This function calculates the message-digest of a buffer,
```

Comment:
Blocker: This description is neither brief nor clear. This is sufficient:

> This function calculates the message digest of a buffer in one shot.

Reply:
I think this is too short, lacking the information on how the concrete algorithm is specified.

Reply:
What else would you add? As with any information needed by a function, the concrete algorithm is specified through one of the parameters. In this case the parameter and the way it conveys the algorithm are obvious.

Reply:
I think it's worth mentioning somehow that here you specify the algorithm in the parameters, whereas the other functions deal with contexts that have been bound to specific algorithms at setup time, before the call performing the actual operation happens. 
I'd be happy with a combination like `This function calculates the message-digest of a buffer with respect to a configurable message-digest algorithm in a single call.`

Reply:
Updated